src/routes/oath.ts

```diff
@@ -22,20 +21,49 @@ const isAuthorized = (userId: string) =>{
     return tokenStore[userId] ? true : false;
 };
 
+// Get new access token from Hubspot
+// Return: Access token object or error object.
+// Access token object: {"refresh_token", "access_token", "expires_in"}
+// Error object: {"message"}
+const getHubspotAccessToken = async (refreshToken: string) => {
+	// return early if missing some env variables
+	if (!HUBSPOT_CLIENT_ID || !HUBSPOT_CLIENT_SECRET) {
+		return {message: "ERROR: Hubspot client id or secret missing."};
+	}
+	try {
+		const url = `https://api.hubapi.com/oauth/v1/token`;
+		const resp = await axios.post(url, null, {
+			params: {
+				grant_type: "refresh_token",
+				client_id: HUBSPOT_CLIENT_ID,
+				client_secret: HUBSPOT_CLIENT_SECRET,
+				refresh_token: refreshToken
+			},
+			headers: {
+				Content_type: "application/x-www-form-urlencoded",
+				charset: "utf-8"
+			}
+		});
+		return resp.data;
+	} catch(e) {
+		console.log(e);
```

Comment:
I guess this would be a good place for me to add the redirect to the install page? It looks like this is the place the code would come if we are unsuccessfull in getting a new access token.

Reply:
I think at line 47 would be the place, catch is for if anything inside the try{} throws an error it will print instead of ending the server.